lib/bumper.js

```diff
@@ -30,65 +73,144 @@ class Bumper {
     if (!this.config.isPr) {
       logger.log(`${pkgJson.name}: Not a PR build, skipping check`)
       return Promise.resolve()
-    } else {
-      return this.getOpenPrScope()
-        .then((scope) => {
-          logger.log(`Found a ${scope} bump for the current PR`)
-        })
     }
+
+    return this._getOpenPrScope()
+      .then((scope) => {
+        logger.log(`Found a ${scope} bump for the current PR`)
+      })
   }
 
+  // ====================================================================================
+  // Private Methods
+  // ====================================================================================
+
   /**
-   * Bump the version based on the last merged PR's version-bump comment
-   * @returns {Promise} a promise resolved with the results of the push
+   * Bump the version in package.json with the given scope
+   * @param {PrInfo} info - the pr info
+   * @param {String} pkgJsonFile - the full file path to the package.json to bump
+   * @returns {PrInfo} the pr info object passed in
    */
-  bump () {
-    const vcs = this.vcs
-    const config = this.config
-    return this.getMergedPrInfo()
-      .then((info) => {
-        return utils.bumpVersion(info, 'package.json')
-      })
-      .then((info) => {
-        return utils.prependChangelog(info, 'CHANGELOG.md')
-      })
+  _bumpVersion (info, pkgJsonFile) {
+    const v = versiony.from(pkgJsonFile).indent(' '.repeat(2))
+    switch (info.scope) {
+      case 'patch':
+        v.patch()
+        break
+
+      case 'minor':
+        v.minor().patch(0)
+        break
+
+      case 'major':
+        v.newMajor()
+        break
+
+      default:
+        throw new Error(`pr-bumper: Invalid scope [${info.scope}]`)
+    }
+
+    const versionInfo = v.to(pkgJsonFile).end()
+    info.version = versionInfo.version
+    return info
+  }
+
+  /**
+   * Commit the changed files 'package.json' and 'CHANGELOG.md'
+   *
+   * @returns {Promise} - a promise resolved with the results of the git commands
+   */
+  _commitChanges () {
+    const buildNumber = this.config.ci.buildNumber
+
+    return this.ci.setupGitEnv()
       .then(() => {
-        return utils.commitChanges(config)
+        // TODO: make this more configurable? (what's being bumped that is)
+        return this.ci.add(['package.json', 'CHANGELOG.md'])
       })
       .then(() => {
-        return vcs.push()
+        return this.ci.commit('Automated version bump [ci skip]', `From CI build ${buildNumber}`)
       })
```

Comment:
Again, I'm not really big on the single-line approach there, it makes it less clear to me what's going on.